### PR TITLE
Doc: More notes for developers adding new PerformanceElements

### PR DESCRIPTION
**Pull request by nielsmh**
I'm unsure about the formatting of the comments, they probably should have some Doxygen formatting codes added.

### CODE REVIEW

**Review by LordAro — DISMISSED** [review]
Reads fine, though i can't vouch for the correctness/usefulness

File: src/framerate_gui.cpp [hunk 1]
```diff
@@ -51,8 +52,10 @@ namespace {
 		/** Start time for current accumulation cycle */
 		TimingMeasurement acc_timestamp;
 
+		/** Initialize a data element with an expected collection rate */
```
Comment (LordAro) [inline src/framerate_gui.cpp]: Any chance of documenting the parameters too? :)

File: src/framerate_type.h [hunk 2]
```diff
@@ -18,9 +23,9 @@
  *
  * There are several places that depend on the order and number of elements in this enum.
  * When adding a new element, make sure to add corresponding new items to all other locations too:
- *  - Translations in english.txt, starting at both STR_FRAMERATE_GAMELOOP and STR_FRAMETIME_CAPTION_GAMELOOP
+ *  - Translations in @c english.txt, starting at both @c STR_FRAMERATE_GAMELOOP and @c STR_FRAMETIME_CAPTION_GAMELOOP
```
Comment (LordAro) [inline src/framerate_type.h]: usually use `\c` for short fields

File: src/framerate_type.h [hunk 3]
```diff
@@ -7,6 +7,11 @@
 * See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
 */
 
+/** @file framerate_type.h
```
Comment (LordAro) [inline src/framerate_type.h:10]: Multiline doxygen comments are usually have the opening `/**` on their own. I'd merge this line and the below as well, looking at the other files

File: src/framerate_type.h [hunk 4]
```diff
@@ -7,6 +7,11 @@
...
+ * Types for recording game performance data.
+ * @see framerate_gui.cpp for implementation
```
Comment (LordAro) [inline src/framerate_type.h:34]: `@see` probably not necessary, it's mostly implied

File: src/framerate_type.h [hunk 5]
```diff
@@ -7,6 +7,11 @@
...
+ */
+
 #ifndef FRAMERATE_GUI_H
```
Comment (LordAro) [inline src/framerate_type.h]: Not really this PR, but... this does not match the filename

**Comment by dorobouNeko** [comment]
I was making a step by step instruction but now that i see this I'll just leave it here and hope it can help and perhaps incorporate some of it in the doc. Only for the PerformanceAccumulator. If it's too badly written just ignore it.
[add-new-measure.txt](https://github.com/OpenTTD/OpenTTD/files/2523773/add-new-measure.txt)

**Comment by nielsmh** [comment]
I might rework things a bit, put a semi-step by step guide in the file comment.

File: src/framerate_type.h (hunk 4 above)
Comment (nielsmh) [reply src/framerate_type.h:34]: I thought it was annoying to navigate from one to the other, which is why I added it :)

**Review by michicc — APPROVED** [review]

**Pull request merged**